stdlib/contrib/bonitoo-io/hex/hex.go

```diff
@@ -0,0 +1,153 @@
+package hex
+
+import (
+	"context"
+	goHex "encoding/hex"
+	"strconv"
+	"strings"
+
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/internal/errors"
+	"github.com/influxdata/flux/runtime"
+	"github.com/influxdata/flux/semantic"
+	"github.com/influxdata/flux/values"
+)
+
+const (
+	conversionArg = "v"
+	pkgName       = "contrib/bonitoo-io/hex"
+)
+
+func init() {
+	runtime.RegisterPackageValue(pkgName, "string", stringConv)
+	runtime.RegisterPackageValue(pkgName, "int", intConv)
+	runtime.RegisterPackageValue(pkgName, "uint", uintConv)
+	runtime.RegisterPackageValue(pkgName, "bytes", bytesConv)
+}
+
+var (
+	convIntType    = runtime.MustLookupBuiltinType(pkgName, "int")
+	convUintType   = runtime.MustLookupBuiltinType(pkgName, "uint")
+	convStringType = runtime.MustLookupBuiltinType(pkgName, "string")
+	convBytesType  = runtime.MustLookupBuiltinType(pkgName, "bytes")
+)
+
+var errMissingArg = errors.Newf(codes.Invalid, "missing argument %q", conversionArg)
+
+var stringConv = values.NewFunction(
```

Comment:
A pattern that I like to do when it comes to functions is this.

```
func init() {
    runtime.RegisterPackageValue(pkgName, "string",
        values.NewFunction(
            "string",
            runtime.MustLookupBuiltinType(pkgName, "string"),
            func(ctx context.Context, args values.Object) (values.Value, error) {
                return interpreter.DoFunctionCallContext(String, ctx, args)
            },
        ),
    )
}

func String(ctx context.Context, args interpreter.Arguments) (values.Value, error) {
    // implement as normal
}
```

Reply:
I have chosen a wrong existing file to copy the pattern from. It is better your way.